From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <nonplay@bxteam.org>
Date: Sun, 27 Apr 2025 14:24:19 +0300
Subject: [PATCH] Raytrace AntiXray SDK integration

Integration with Imanity Software's Raytrace AntiXray for better use of this plugin

Original project: https://github.com/Imanity-Software/raytrace-antixray-spigot-sdk

diff --git a/net/minecraft/server/level/ServerPlayerGameMode.java b/net/minecraft/server/level/ServerPlayerGameMode.java
index 96a023f80db0596629cbd0c1a5c6df273ac26d6b..28c69f61e476b138a38ae6eb9b5ae06b3bb818d7 100644
--- a/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -326,6 +326,12 @@ public class ServerPlayerGameMode {
                 this.abortDestroyBlock(pos, sequence);
             }
         }
+        // DivineMC start - Imanity: Raytrace AntiXraySDK integration
+        dev.imanity.antixray.sdk.AntiXrayAdapter adapter = dev.imanity.antixray.sdk.AntiXraySDK.getAdapter();
+        if (adapter != null) {
+            adapter.callPlayerLeftClickBlock(this.level.getWorld(), this.player.getBukkitEntity(), pos.getX(), pos.getY(), pos.getZ());
+        }
+        // DivineMC end - Imanity: Raytrace AntiXraySDK integration
         this.level.chunkPacketBlockController.onPlayerLeftClickBlock(this.level, pos, action, direction, maxY, sequence); // Paper - Anti-Xray
     }
 
diff --git a/net/minecraft/world/level/Level.java b/net/minecraft/world/level/Level.java
index 70aa16cb21a9d1123db75310adbf280198c3b625..01daa5e034f6b34ae466f6a2f9ee4b39c7c25be1 100644
--- a/net/minecraft/world/level/Level.java
+++ b/net/minecraft/world/level/Level.java
@@ -1143,6 +1143,12 @@ public abstract class Level implements LevelAccessor, AutoCloseable, ca.spottedl
         }
         // CraftBukkit end - capture blockstates
         BlockState oldState = chunk.setBlockState(pos, blockState, updateFlags);
+        // DivineMC start - Imanity: Raytrace AntiXraySDK integration
+        dev.imanity.antixray.sdk.AntiXrayAdapter adapter = dev.imanity.antixray.sdk.AntiXraySDK.getAdapter();
+        if (adapter != null) {
+            adapter.callBlockChange(world, pos.getX(), pos.getY(), pos.getZ(), blockState.getBukkitMaterial());
+        }
+        // DivineMC end - Imanity: Raytrace AntiXraySDK integration
         this.chunkPacketBlockController.onBlockChange(this, pos, blockState, oldState, updateFlags, updateLimit); // Paper - Anti-Xray
         if (oldState == null) {
             // CraftBukkit start - remove blockstate if failed (or the same)
